tools/scripts/deploy.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/sh
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2020 Tomer Eliyahu (Intel)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
+
+scriptdir="$(cd "${0%/*}" && pwd)"
+rootdir="${scriptdir%/*/*}"
+
+dbg() {
+    [ "$VERBOSE" = "true" ] && echo "$@"
+}
+
+err() {
+    printf "\033[1;31m%s\n\033[0m" "$@"
+}
+
+success() {
+    printf "\033[1;32m%s\n\033[0m" "$@"
+}
+
+run() {
+    dbg "$*"
+    "$@" || exit $?
+}
+
+
+usage() {
+    echo "usage: $(basename "$0") [-hvdb] [--target <user@target>] [--proxy <user@proxy>]"
+}
+
+main() {
+
+    if ! OPTS=$(getopt -o 'hvdb:' --long help,verbose,deploy,build-dir:,target:,proxy: -n 'parse-options' -- "$@"); then
+        err "Failed parsing options." >&2
+        usage
+        exit 1 
+    fi
+
+    eval set -- "$OPTS"
+
+    while true; do
+        case "$1" in
+            -v | --verbose)       VERBOSE=true; shift ;;
+            -h | --help)          usage; exit 0 ;;
+            -d | --deploy)        DEPLOY=true; shift ;;
+            -b | --build-dir)     BUILD_DIR="$2"; shift; shift ;;
```

Comment:
This is only used for two things:

- The install directory, i.e. `${BUILD_DIR}/install`.
- The location of the temporary tarball.

I think it would be more appropriate to instead pass the install directory, and use an actual temporary file (i.e., a `$(mktemp --tmpdir prplmesh-XXXX.tar.gz)`) for the tarball location. But that would of course mean we have to delete it as well. Hm, I'm not sure...

Anyway, can be changed later.

Reply:
The reason I didn't put the tar in a random location is because of the `--deploy` flag - you can use the script only to create `prplmesh.tar.gz`, so it helps if it is in a known location :)